baseapp/msg_service_router.go

```diff
@@ -29,12 +29,17 @@ func NewMsgServiceRouter() *MsgServiceRouter {
 }
 
 // MsgServiceHandler defines a function type which handles Msg service message.
-type MsgServiceHandler = func(ctx sdk.Context, req sdk.MsgRequest) (*sdk.Result, error)
+type MsgServiceHandler = func(ctx sdk.Context, req sdk.Msg) (*sdk.Result, error)
 
-// Handler returns the MsgServiceHandler for a given query route path or nil
+// Handler returns the MsgServiceHandler for a given msg or nil if not found.
+func (msr *MsgServiceRouter) Handler(msg sdk.Msg) MsgServiceHandler {
+	return msr.routes[sdk.MsgTypeURL(msg)]
+}
+
+// HandlerByName returns the MsgServiceHandler for a given query route path or nil
 // if not found.
-func (msr *MsgServiceRouter) Handler(methodName string) MsgServiceHandler {
-	return msr.routes[methodName]
+func (msr *MsgServiceRouter) HandlerByName(msgName string) MsgServiceHandler {
```

Comment:
If this method takes a message name starting with a `/`, then I think it should be called `HandlerByTypeURL` to clarify that it needs a typeURL and not a message name.